torch/_dynamo/variables/builder.py

```diff
@@ -1902,10 +1902,20 @@ def wrap_to_fake_tensor_and_record(
                 )
 
         tx.output.tracing_context.tensor_to_context[e] = symbolic_context
-        tx.output.input_source_to_sizes_strides[source] = {
-            "size": fake_e.size(),
-            "stride": fake_e.stride(),
-        }
+        if is_sparse_any(fake_e):
+            # TODO: for TensorGuards, this eventually may need more
+            #       fields for the size/stride of any other constituents
+            tx.output.input_source_to_sizes_strides[source] = {
+                "size": fake_e.size(),
+                "stride": (1,) * fake_e.ndim,
```

Comment:
Could we use `"stride": ()` or `"stride": None`? In the case of sparse tensors, `stride` should never be accessed as its content has no meaning. Using invalid stride here would allow catching silent misuses of the stride data set here.

Reply:
The stride () resulted in a segfault, I suspect because the tensor guards code was accessing the individual fields. let me go a bit deeper into that code to see what else would work